2-WebApp-graph-user/2-6-BFF-Proxy/README.md

```diff
@@ -0,0 +1,417 @@
+---
+page_type: sample
+name: React SPA with ASP.NET Core backend calling Microsoft Graph using the backend for frontend proxy architecture
+description: A React single-page application with an ASP.NET Core backend authenticating users and calling the Microsoft Graph API using the backend for frontend proxy architecture
+languages:
+ - csharp
+ - javascript
+products:
+ - azure-active-directory
+ - microsoft-identity-web
+urlFragment: active-directory-aspnetcore-webapp-openidconnect-v2
+extensions:
+- services: ms-identity
+- platform: .NET Core
+- endpoint: AAD v2.0
+- level: 300
+- client: ASP.NET Core web app
+- service: Microsoft Graph
+---
+
+# React SPA with ASP.NET Core backend calling Microsoft Graph using the backend for frontend (BFF) proxy architecture
+
+[![Build status](https://identitydivision.visualstudio.com/IDDP/_apis/build/status/AAD%20Samples/.NET%20client%20samples/ASP.NET%20Core%20Web%20App%20tutorial)](https://identitydivision.visualstudio.com/IDDP/_build/latest?definitionId=XXX)
+
+* [Overview](#overview)
+* [Scenario](#scenario)
+* [Prerequisites](#prerequisites)
+* [Setup the sample](#setup-the-sample)
+* [Explore the sample](#explore-the-sample)
+* [Troubleshooting](#troubleshooting)
+* [About the code](#about-the-code)
+* [Next Steps](#next-steps)
+* [Contributing](#contributing)
+* [Learn More](#learn-more)
+
+## Overview
+
+This sample demonstrates a React single-page application (SPA) with an ASP.NET Core backend that authenticates users and calls the Microsoft Graph API using the backend for frontend (BFF) proxy architecture. In this architecture, access tokens are retrieved and stored within the secure backend context, and the client side JavaScript application, which is served by the ASP.NET web app, is only indirectly involved in the authN/authZ process by routing the token and API requests to the backend. The trust between the frontend and backend is established via a secure cookie upon successful sign-in.
```

Comment:
reminder: link to ietf-draft for "BFF proxy architecture" https://datatracker.ietf.org/doc/html/draft-ietf-oauth-browser-based-apps#section-6.2

Reply:
done